### ISSUE TITLE
Confusion about level 2 files

### ISSUE THREAD

**Issue by bpbond** [JGCRI/gcamdata#615]
Hi @pralitp -

Many people (including me) are pretty fuzzy about how to handle the `get_logit_fn_tables` calls and associated `batch_` files, and folks (including me) are finding the documentation at https://github.com/JGCRI/gcamdata/wiki/Name-That-Function unclear.

Do you have time to create a dedicated Level 2 chunk wiki page? It would detail the parts to avoid (`get_logit_function_tables`) a little more clearly, demonstrate LEVEL2_DATA_NAMES usage (e.g. https://github.com/JGCRI/gcamdata/pull/609#discussion_r131474746 ) and generally provide better and clearer guidance.

On a related note, at some point you and I should discuss some late-stage issues, e.g. verifying xml files (presence/absence/contents).

Thanks!

**Comment by bpbond** [comment]
@rynacui @abigailsnyder @russellhz @aarony2j I opened this and Pralit, very reasonably, asked for specific examples of where the existing documentation is unclear or confusing. If you get a chance and have one to provide to him, either in this issue or in person, please do. Thanks.

**Comment by bpbond** [comment]
Here's one question: every time I remove a `get_logit_fn_tables` call, should I be adding a "logit.type" column to `write_to_all_regions`? Should I also, every time, be adding a `add_logit_tables_xml` call in the associated xml-generating file? This wasn't clear to me.

**Comment by pralitp** [comment]
The answer is yes to both questions.  I can add something like the following to the wiki if you find it helpful:
The reason being is are kicking all the work that would have been done by `get_logit_fn_tables` down the road into the "batch chunks" where that same work is essentially being done with the call to `add_logit_tables_xml`.  However the call to `write_to_all_regions` would want to drop the `logit.type` column but we need to maintain it so that it is still available to use down the road when `add_logit_tables_xml` is finally called.

**Comment by bpbond** [comment]
Ah, OK, thanks!

We'll need to go back/through the xml-generating code, I suspect. I definitely have _not_ being doing this to date.

**Comment by pralitp** [comment]
I just did a quick check on the current master and found just one case: `ag_For_Past_bio_base_IRR_MGMT.xml`

**Comment by bpbond** [comment]
How should I handle lines 321-335 in `L202.an_input.R` (aglu/level2)? Does a version of this code have to go into the associated xml-generating file in the new system?

```R
for( curr_table in names( L202.SectorLogitTables_in ) ) {
    if( curr_table != "EQUIV_TABLE" ) {
        L202.SectorLogitTables_in[[ curr_table ]]$data <- subset( L202.SectorLogitTables_in[[ curr_table ]]$data,
            !region %in% no_aglu_regions )
    }
}
L202.Supplysector_in <- subset( L202.Supplysector_in, !region %in% no_aglu_regions )
for( curr_table in names( L202.SubsectorLogitTables_in ) ) {
    if( curr_table != "EQUIV_TABLE" ) {
        L202.SubsectorLogitTables_in[[ curr_table ]]$data <- subset( L202.SubsectorLogitTables_in[[ curr_table ]]$data,
            !region %in% no_aglu_regions &
            !vecpaste( L202.SubsectorLogitTables_in[[ curr_table ]]$data[ c( reg, supp, subs ) ] ) %in%
             vecpaste( L202.no_ddgs_regions_subs[ c( reg, supp, subs ) ] ) )
    }
}
```

**Comment by pralitp** [comment]
I think all of the code in the for loops can just be deleted.  However:
`L202.Supplysector_in <- subset( L202.Supplysector_in, !region %in% no_aglu_regions )` should stay.  And no, this sort of code will not need to be replicated in the xml-generating chunk.

Essentially we needed to strip out `no_aglu_regions` from all tables but because in our old strategy we created `L202.SectorLogitTables_in` from `L202.Supplysector_in` so early in the process that means any adjustments made to `L202.Supplysector_in` after would have to get replicated in the `L202.SectorLogitTables_in` tables as well.

We side step this entire problem in the new strategy because we do not generate `L202.SectorLogitTables_in` from `L202.Supplysector_in` until the very last moment -- just before we create the xml via `add_logit_tables_xml `.

Does that make sense?
